internal/modgen/modgen.go

```diff
@@ -0,0 +1,420 @@
+package modgen
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"strconv"
+
+	"github.com/tetratelabs/wazero/internal/leb128"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// Gen generates a pseudo random compilable module based on `seed`. The size of
```

Comment:
I think the docs are better spent on what's generated vs the relationship between the size and the seed. I suspect we'll pull out any indirect relationship between seed and size at some point, and meanwhile knowing what this results in could be a better primary choice to talk about.

Ex. the seed's most interesting and important thing is that it allows deterministic generation. I would even go so far to say not random, but seed-deterministic maybe